x-pack/plugins/security_solution/server/endpoint/services/actions/utils/fetch_action_requests.ts

```diff
@@ -0,0 +1,169 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type {
+  SearchRequest,
+  QueryDslQueryContainer,
+  QueryDslBoolQuery,
+  SearchTotalHits,
+} from '@elastic/elasticsearch/lib/api/types';
+import { fromKueryExpression, toElasticsearchQuery } from '@kbn/es-query';
+import type { ElasticsearchClient } from '@kbn/core-elasticsearch-server';
+import type { Logger } from '@kbn/logging';
+import { stringify } from '../../../utils/stringify';
+import { getDateFilters } from '../..';
+import { ENDPOINT_ACTIONS_INDEX } from '../../../../../common/endpoint/constants';
+import { catchAndWrapError } from '../../../utils';
+import type { LogsEndpointAction } from '../../../../../common/endpoint/types';
+import type {
+  ResponseActionAgentType,
+  ResponseActionsApiCommandNames,
+  ResponseActionType,
+} from '../../../../../common/endpoint/service/response_actions/constants';
+
+export interface FetchActionRequestsOptions {
+  esClient: ElasticsearchClient;
+  logger: Logger;
+  from?: number;
+  size?: number;
+  startDate?: string;
+  endDate?: string;
+  agentTypes?: ResponseActionAgentType[];
+  commands?: ResponseActionsApiCommandNames[];
+  elasticAgentIds?: string[];
+  userIds?: string[];
+  unExpiredOnly?: boolean;
+  types?: ResponseActionType[];
+}
+
+interface FetchActionRequestsResponse {
+  data: LogsEndpointAction[];
+  total: number;
+  from: number;
+  size: number;
+}
+
+/**
+ * Fetches a list of Action Requests from the Endpoint action request index (not fleet)
+ * @param logger
+ * @param agentTypes
+ * @param commands
+ * @param elasticAgentIds
+ * @param esClient
+ * @param endDate
+ * @param from
+ * @param size
+ * @param startDate
+ * @param userIds
+ * @param unExpiredOnly
+ * @param types
+ */
+export const fetchActionRequests = async ({
```

Comment:
I extracted the fetching of action requests (list) into its own utility here so that it can be reused